### PR TITLE
Closes #244 Used derive_vars_cat() to derive AVALCA1N and AVALCAT1, replacing look-up table with derive_vars_merged()

### LINKED ISSUE #244: Update ADBCVA vignette, template to use new {admiral} AVALCAT derivation function
### Background Information

Currently there is nothing specific in {admiral} for AVALCAT derivation so ADBCVA template and vignette just go about it using tidyverse tools. This will change once the functionality is added to {admiral} - [see here](https://github.com/pharmaverse/admiral/issues/2480).

### Definition of Done

Update template and vignette to use new {admiral} function.

### PR DESCRIPTION
**Pull request by steventing12**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update).
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files.
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions]
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately.
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralophtha (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralophtha site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralophtha/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`.
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`.
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@bundfussr pls can I ask for your help here - any idea why these checks are failing? As far as I can see the function is listed in the imports so not sure what is happening.

**Comment by bundfussr** [comment]
> @bundfussr pls can I ask for your help here - any idea why these checks are failing? As far as I can see the function is listed in the imports so not sure what is happening.

I think the issue in the failing workflows is that the installation of admiral fails. However, I don't know why it fails for some workflows but works for the others. It seems that it works for those workflows where staged.dependencies is installed in the "Run staged dependencies" step. @cicdguy , any ideas?

File: vignettes/adbcva.Rmd [hunk 1]
```diff
@@ -51,7 +51,7 @@ library(admiralophtha)
 
 ## Initial set up of ADBCVA {#setup}
 
-As with all BDS ADaM datasets, one should start from the OE SDTM, where only the BCVA records are of interest. For the purposes of the next two sections, we shall be using the `{admiral}` OE and ADSL test data. We will also require a lookup table for the mapping of parameter codes.
+As with all BDS ADaM datasets, one should start from the OE SDTM, where only the BCVA records are of interest. For the purposes of the next two sections, we shall be using the `{admiral}` OE and ADSL test data. We will also require a lookup table for the mapping of parameter codes. A SBCVA and FBCVA definition list which can be sourced from a separate R script is included as well.
```
Comment (manciniedoardo) [inline vignettes/adbcva.Rmd:54]:
```suggestion
As with all BDS ADaM datasets, one should start from the OE SDTM, where only the BCVA records are of interest. For the purposes of the next two sections, we shall be using the `{admiral}` OE and ADSL test data. We will also require a lookup table for the mapping of parameter codes. An SBCVA and FBCVA definition expression is created first - this expression can also be stored within another (sourced) program.
```

File: vignettes/adbcva.Rmd [hunk 2]
```diff
@@ -185,74 +232,13 @@ adbcva <- admiralophtha_adbcva %>%
...
-Some ophthalmology studies may desire to subdivide BCVA records according to which Snellen category they fall into (eg, 20/320, 20/100, 20/20 etc). This is best done through the use of `AVALCATx`/`AVALCAxN` variable pairs. Currently, `{admiralophtha}` does not provide specific functionality to create `AVALCATx`/`AVALCAxN` pairs, although this may be included in future releases of the package. With the current toolset, the suggested approach to derive such variables is to:
-
-* Create a lookup table which assigns numeric equivalents (i.e. `AVALCAxN`) to Snellen categories.
-* Create a format function to map each `AVAL` to a numeric category.
-* Add `AVALCAxN` through a mutate statement using the format function.
-* Add `AVALCATx` using `derive_vars_merged` in combination with the lookup table.
+Some ophthalmology studies may desire to subdivide BCVA records according to which Snellen category they fall into (eg, 20/320, 20/100, 20/20 etc). The {admiral} AVALCAT derivation function `derive_vars_cat` can be used to derive `AVALCA1N` and `AVALCAT1` based on `PARAMCD` and `condition` in the SBCVA and FBCVA definition list.
```
Comment (manciniedoardo) [inline vignettes/adbcva.Rmd:235]:
```suggestion
Some ophthalmology studies may desire to subdivide BCVA records according to which Snellen category they fall into (eg, 20/320, 20/100, 20/20 etc). The `{admiral}` AVALCAT derivation function `derive_vars_cat` can be used to derive `AVALCA1N` and `AVALCAT1` based on `PARAMCD` and `condition` in the SBCVA and FBCVA definition list.
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @steventing12 this looks great! Left some comments, but once addressed please request review again and we can merge in.

As a general reminder, please ensure functions and "{admiral}`" are always enclosed with backticks - this ensures that within the pkgdown website they are rendered as dynamic links to their respective documentation pages.

**Comment by steventing12** [comment]
> Thanks @steventing12 this looks great! Left some comments, but once addressed please request review again and we can merge in.
> 
> As a general reminder, please ensure functions and "{admiral}`" are always enclosed with backticks - this ensures that within the pkgdown website they are rendered as dynamic links to their respective documentation pages.

Hi Edoardo, I have committed your suggestions and checks are completed. Please see if all is fine now, thanks.

File: NEWS.md [hunk 3]
```diff
@@ -22,6 +22,12 @@
   explicitly quoting key variables such as `STUDYID` and `USUBJID`. For instance, a line such as `by_vars = exprs(STUDYID, USUBJID)`
   would be replaced by `by_vars = get_admiral_option("subject_keys")` (#226).
 
+# admiralophtha 1.1.0
+
+## Updates to Templates
+
+- Within the ADBCVA template, updated `AVALCA1N` and `AVALCAT1` derivations to use new function `admiral::derive_vars_cat()`. (#244)
```
Comment (manciniedoardo) [inline NEWS.md:29]: @steventing12 this shouldn't be under admiralophtha 1.1.0 as that version has already released - it should be under 1.0.0.9007

File: vignettes/adbcva.Rmd [hunk 4]
```diff
@@ -185,74 +232,13 @@ adbcva <- admiralophtha_adbcva %>%
...
-Some ophthalmology studies may desire to subdivide BCVA records according to which Snellen category they fall into (eg, 20/320, 20/100, 20/20 etc). This is best done through the use of `AVALCATx`/`AVALCAxN` variable pairs. Currently, `{admiralophtha}` does not provide specific functionality to create `AVALCATx`/`AVALCAxN` pairs, although this may be included in future releases of the package. With the current toolset, the suggested approach to derive such variables is to:
-
-* Create a lookup table which assigns numeric equivalents (i.e. `AVALCAxN`) to Snellen categories.
-* Create a format function to map each `AVAL` to a numeric category.
-* Add `AVALCAxN` through a mutate statement using the format function.
-* Add `AVALCATx` using `derive_vars_merged` in combination with the lookup table.
+Some ophthalmology studies may desire to subdivide BCVA records according to which Snellen category they fall into (eg, 20/320, 20/100, 20/20 etc). The `{admiral}` AVALCAT derivation function `derive_vars_cat` can be used to derive `AVALCA1N` and `AVALCAT1` based on `PARAMCD` and `condition` in the SBCVA and FBCVA definition list.
```
Comment (manciniedoardo) [inline vignettes/adbcva.Rmd:235]:
```suggestion
Some ophthalmology studies may desire to subdivide BCVA records according to which Snellen category they fall into (eg, 20/320, 20/100, 20/20 etc). The `{admiral}` AVALCAT derivation function `derive_vars_cat` can be used to derive `AVALCA1N` and `AVALCAT1` based on `PARAMCD` and `condition` in the SBCVA and FBCVA definition expression.
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
@steventing12 couple more comments sorry!

File: NEWS.md (hunk 3 above)
Comment (steventing12) [reply NEWS.md:29]: Updated, please check again

**Comment by steventing12** [comment]
> @steventing12 couple more comments sorry!

No worries, updates pushed

File: NEWS.md [hunk 5]
```diff
@@ -13,6 +13,12 @@
   explicitly quoting key variables such as `STUDYID` and `USUBJID`. For instance, a line such as `by_vars = exprs(STUDYID, USUBJID)`
   would be replaced by `by_vars = get_admiral_option("subject_keys")` (#226).
 
+# admiralophtha 1.1.0
+
+## Updates to Templates
+
+- Replaced look-up table with derive_vars_merged() in adbcva_crtflag dataframe with derive_vars_cat() to derive AVALCA1N and AVALCAT1 in ADBCVA template using new SBCVA and FBCVA definition list (#244)
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply NEWS.md]: @steventing12 Just to add, I don't see an entry under updates to documentation here?

[the remark this answers is not in the document]
Comment (manciniedoardo) [reply NEWS.md]: > It's at the top under # admiralophtha 1.1.0.9007
> 
> ## Updates to Templates
> * Within the ADBCVA template, updated `AVALCA1N` and `AVALCAT1` derivations to use new function `admiral::derive_vars_cat()`. ([Update ADBCVA vignette, template to use new {admiral} AVALCAT derivation function #244](https://github.com/pharmaverse/admiralophtha/issues/244))

That's the update to the template, but you also updated the User Guide (vignette) so need to add a separate item for that too.

File: NEWS.md [hunk 6]
```diff
@@ -1,5 +1,13 @@
 # admiralophtha 1.1.0.9007
 
+## Updates to Templates
+
+- Within the ADBCVA template, updated `AVALCA1N` and `AVALCAT1` derivations to use new function `admiral::derive_vars_cat()`. (#244)
+
+## Updates to Documentation
+
+- Within the ADBCVA template, updated `AVALCA1N` and `AVALCAT1` derivations to use new function `admiral::derive_vars_cat()`. (#244)
```
Comment (manciniedoardo) [inline NEWS.md:9]:
```suggestion
- Within the ADBCVA vignette, updated `AVALCA1N` and `AVALCAT1` derivations to use new function `admiral::derive_vars_cat()`. (#244)
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Typo fix requested

**Comment by manciniedoardo** [comment]
> Typo fix requested

Went ahead and updated myself so that we can merge 😄

**Review by manciniedoardo — APPROVED** [review]
Thanks @steventing12 great job here!

**Comment by steventing12** [comment]
> > Typo fix requested
> 
> Went ahead and updated myself so that we can merge 😄

Thanks Edoardo!
